mycore-iview2/src/main/java/org/mycore/iview2/frontend/MCRPDFTools.java

```diff
@@ -30,60 +30,127 @@
+    public static BufferedImage getThumbnail(Path pdfFile, int thumbnailSize, boolean centered) throws IOException {
+        return getThumbnail(-1, pdfFile, false);
+    }
+
+    /**
+     * This method returns a Buffered Image as thumbnail if an initial page was set,
+     * it will be return - if not the first page
+     *
+     * @param thumbnailSize - the size: size = max(width, height) 
+     *                        a size &lt; 0 will return the original size and centered parameter will be ignored
+     * @param pdfFile       - the file from which the thumbnail will be taken
+     * @param centered      - if true, a square (thumbnail with same width and
+     *                      height) will be returned
+     * @return a BufferedImage as thumbnail
+     *
+     * @throws IOException
+     */
+    public static BufferedImage getThumbnail(int thumbnailSize, Path pdfFile, boolean centered) throws IOException {
         InputStream fileIS = Files.newInputStream(pdfFile);
         try (PDDocument pdf = PDDocument.load(fileIS)) {
             PDFRenderer pdfRenderer = new PDFRenderer(pdf);
-            BufferedImage level1Image = pdfRenderer.renderImage(0);
-            int imageType = BufferedImage.TYPE_INT_ARGB;
-
-            if (!centered) {
+            final PDPage page = resolveOpenActionPage(pdf);
+            BufferedImage level1Image = pdfRenderer.renderImage(pdf.getPages().indexOf(page));
+            if (thumbnailSize < 0) {
                 return level1Image;
             }
-            final double width = level1Image.getWidth();
-            final double height = level1Image.getHeight();
-            LOGGER.info("new PDFBox: {}x{}", width, height);
-            LOGGER.info("temporary image dimensions: {}x{}", width, height);
-            final int newWidth = width < height ? (int) Math.ceil(thumbnailSize * width / height) : thumbnailSize;
-            final int newHeight = width < height ? thumbnailSize : (int) Math.ceil(thumbnailSize * height / width);
-            //if centered make thumbnailSize x thumbnailSize image
-            final BufferedImage bicubic = new BufferedImage(centered ? thumbnailSize : newWidth,
-                centered ? thumbnailSize : newHeight, imageType);
-            LOGGER.info("target image dimensions: {}x{}", bicubic.getWidth(), bicubic.getHeight());
-            final Graphics2D bg = bicubic.createGraphics();
-            try {
-                bg.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_BICUBIC);
-                int x = centered ? (thumbnailSize - newWidth) / 2 : 0;
-                int y = centered ? (thumbnailSize - newHeight) / 2 : 0;
-                if (x != 0 && y != 0) {
-                    LOGGER.warn("Writing at position {},{}", x, y);
+            return scalePage(thumbnailSize, centered, level1Image);
+        }
+    }
+
+    private static BufferedImage scalePage(int thumbnailSize, boolean centered, BufferedImage level1Image) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'scalePage(int, boolean, BufferedImage)' has an NPath complexity of 208](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5425159)